### PR TITLE
Latest rn changes

### PR DESCRIPTION
**Pull request by czakdesign**
Hey @Sfshaza  - the latest changes are in this commit:
Updated file per Shams Review comments 

Dated today - 
/Cz

### CODE REVIEW

File: flutter-for-react-native/index.md [hunk 1]
```diff
@@ -0,0 +1,2411 @@
...
+
+
+
+|Android|iOS|
+|:---:|:--:|
+|<img src="images/state_change_android.gif?raw=true" style="width:300px;" alt="Loading">|<img src="images/state_change_iOS.gif?raw=true" style="width:300px;" alt="Loading">|
+
+<br>
+
+In Flutter, you assign a local variable or function marked `final` with the
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: This is a big Oopsie. We lost a header: Props. I couldn't figure out why this sentece was sitting here, or what the next screenshots were about, until I looked at the original. Please reinstate the Props heading, the (short) paragraph, and the code that follows.

File: flutter-for-react-native/index.md [hunk 2]
```diff
@@ -0,0 +1,2411 @@
...
+const MyApp = TabNavigator(
+  { Home: { screen: HomeScreen }, Notifications: { screen: tabNavScreen } },
+  { tabBarOptions: { activeTintColor: "#e91e63" } }
+);
+```
+
+Flutter provides several specialized widgets for drawer and tab navigation:
+
+* [TabController](https://docs.flutter.io/flutter/material/TabController-class.html)—Coordinates the tab selection between a TabBar and a TabBarView.  
+
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Delete. It makes the final list look disjointed.

File: flutter-for-react-native/index.md [hunk 3]
```diff
@@ -0,0 +1,2411 @@
...
+* [TabBar](https://docs.flutter.io/flutter/material/TabBar-class.html) —Displays a horizontal row of tabs.  
+* [Tab](https://docs.flutter.io/flutter/material/Tab-class.html)—Creates a material design TabBar tab.  
+
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Delete. It makes the final list look disjointed.

File: flutter-for-react-native/index.md [hunk 4]
```diff
@@ -0,0 +1,2411 @@
...
+|Android|iOS|
+|:---:|:--:|
+|<img src="images/navigation_android.gif?raw=true" style="width:300px;" alt="Loading">|<img src="images/navigation_iOS.gif?raw=true" style="width:300px;" alt="Loading">|
+
+<br>
+
+## Gesture detection and touch event handling
+
+To listen for and respond to gestures, Flutter includes taps, drags, and
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Flutter includes taps... => Flutter supports taps...

File: flutter-for-react-native/index.md [hunk 5]
```diff
@@ -0,0 +1,2411 @@
...
+        <View style={styles.center}>
+          <Text>Swipe Horizontally or Vertically</Text>
+        </View>
+      </View>
+    );
+  }
+}
+```
+
+In Flutter, to add a click or press listeners to a widget, you can use buttons
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: => In Flutter, to add a click (or press) listener to a widget, use a button or a touchable widget that has an `onPress:` field. Or, add gesture detection to any widget by wrapping it in a [`GestureDetector`]().

File: flutter-for-react-native/index.md [hunk 6]
```diff
@@ -0,0 +1,2411 @@
...
+
+
+
+|Android|iOS|
+|:---:|:--:|
+|<img src="images/flutter_gestures_android.gif?raw=true" style="width:300px;" alt="Loading">|<img src="images/flutter_gestures_iOS.gif?raw=true" style="width:300px;" alt="Loading">|
+
+<br>
+
+## Making HTTP networking requests
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: => Making HTTP network requests

File: flutter-for-react-native/index.md [hunk 7]
```diff
@@ -0,0 +1,2411 @@
...
+input box and then use the callback to store the value in a variable.
+
+<!-- skip -->
+```JavaScript
+// React Native
+<TextInput
+  placeholder="Enter your Password"
+  onChangeText={password => this.setState({ password })}
+ />
+<Button title
```
Comment (Sfshaza) [inline flutter-for-react-native/index.md]: form => Form

Comment (Sfshaza) [inline flutter-for-react-native/index.md]: If I turned that comma into a hyphen, I apologize. It makes no sense. Please put it back (and delete the comma):

widget-where => widget where

Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Remove the ".." in the URL.

Comment (Sfshaza) [reply flutter-for-react-native/index.md]: (This is a global change.)

Comment (Sfshaza) [inline flutter-for-react-native/index.md]: This is not the same sentence that appears in the staged version. There it says:

In Flutter, we can achieve this animation by using the Dismissible widget and nest the child widgets.

Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Ugh. No.

The child can be a text or image widget. Or something else. Delete the bit that is has to be a Text widget and change it to say, "What appears on the button."  Or something similar.

Comment (Sfshaza) [inline flutter-for-react-native/index.md]: Ditto.

**Comment by Sfshaza** [comment]
I see that your build failed, so I looked:

```
- ./_site/flutter-for-react-native/index.html
  *  internally linking to pub.dartlang.org, which does not exist (line 997)
     <a href="pub.dartlang.org">pub.dartlang.org</a>
```
The link should be https://pub.dartlang.org/

File: flutter-for-react-native/index.md [hunk 8]
```diff
@@ -0,0 +1,2411 @@
...
+ syntax that is AOT-compiled into native.
+* Transcompiles optionally into JavaScript.
+* Supports interfaces and abstract classes.
+
+A few examples of the differences between JavaScript and Dart are described
+ below.
+
+### Entry point
+
+JavaScript doesn't have an entry function—you define the entry point.
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done - reverted back to pre-defined

File: flutter-for-react-native/index.md [hunk 9]
```diff
@@ -0,0 +1,2411 @@
...
+
+<!-- skip -->
+```JavaScript
+// JavaScript
+function main() {
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 10]
```diff
@@ -0,0 +1,2411 @@
...
+}
+// can also be written as
+bool fn() {
+  return true;
+}
+```
+
+Try it out in [DartPad](https://dartpad.dartlang.org/5454e8bfadf3000179d19b9bc6be9918).
+
+For more information, see the documentation on [functions](https://www.dartlang.org/resources/dart-tips/dart-tips-ep-6).
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 11]
```diff
@@ -0,0 +1,2411 @@
...
+<!-- skip -->
+```Dart
+// Dart
+var name; // == null
+int x; // == null
+```
+
+Try it out in the [DartPad](https://dartpad.dartlang.org/57ec21faa8b6fe2326ffd74e9781a2c7).
+
+For more information, see the documentation on [variables](https://www.dartlang.org/resources/dart-tips/dart-tips-ep-3).
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 12]
```diff
@@ -0,0 +1,2411 @@
...
+ You can run Flutter apps in a couple of ways:
+
+ * Use `flutter run` from the project's root directory.
+ * Use the "run" option in an IDE with the Flutter and Dart plugins.
+
+ Your app runs on a connected device, the iOS emulator, or the Android simulator.
+
+For more information, see the Flutter [Getting Started](../get-started/) documentation.
+
+### How do I import widgets??
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: lol - Done!

File: flutter-for-react-native/index.md [hunk 13]
```diff
@@ -0,0 +1,2411 @@
...
+   received, or user input.
+
+The important difference between stateless and stateful widgets is that
+ StatefulWidgets have a State object that stores state data and carries it over
+  across tree rebuilds, so it's not lost.
+
+The widget tree for the previous example consists of 5 widgets: the
+ [MaterialApp](https://docs.flutter.io/flutter/material/MaterialApp-class.html) root widget, and the  [Scaffold](https://docs.flutter.io/flutter/material/Scaffold-class.html), [AppBar](https://docs.flutter.io/flutter/material/AppBar-class.html),   [Center](https://docs.flutter.io/flutter/widgets/Center-class.html), and [Text](https://docs.flutter.io/flutter/widgets/Text-class.html) widgets.
+
+The `MaterialApp` widget wraps a number of widgets that are commonly required
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 14]
```diff
@@ -0,0 +1,2411 @@
...
+
+In Flutter, add a static image to your app using the `AssetImage` class in a
+widget’s build method.
+
+<!-- skip -->
+```Dart
+image: new AssetImage('assets/background.png'),
+```
+
+For more information, see [Adding Assets and Images in Flutter](../assets-and-images/).
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done - globally

File: flutter-for-react-native/index.md [hunk 15]
```diff
@@ -0,0 +1,2411 @@
...
+In Flutter, define a class to create a custom widget and then reuse the
+widget. You can also define and call a function that returns a reusable widget
+as shown in the `build` function in the following example.
+
+<!-- skip -->
+{% prettify dart %}
+
+// Flutter
+class CustomCard extends StatelessWidget {
+  [[highlight]]CustomCard({@requiredthis.index, @required [[/highlight]]
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 16]
```diff
@@ -0,0 +1,2411 @@
...
+
+<br>
+
+## Layouts
+### How do I use widgets to define layout properties?
+
+In React Native, most of the layout can be done with the props that are passed
+to a specific component. For example, you could use the `style` prop on the
+`View` component in order to specify the flexbox properties. To arrange your
+components in a column, you would specify a prop such as: `flexDirection:
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 17]
```diff
@@ -0,0 +1,2411 @@
...
+
+<!-- skip -->
+```Dart
+new Icon(Icons.lightbulb_outline, color: Colors.redAccent)
+```
+
+When using the `Icons` class, make sure to set `uses-material-design: true` in
+the project's `pubspec.yaml` file. This ensures that
+the `MaterialIcons` font, which displays the icons, is included in your app.
+This font is used to display the icons.
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 18]
```diff
@@ -0,0 +1,2411 @@
...
+ a dependency for `cupertino_icons` in your project's 
+ `pubspec.yaml` file.
+<!-- skip -->
+```yaml
+name: my_awesome_application
+dependencies:
+  cupertino_icons: ^0.1.0
+```
+
+To globally customize the colors and styles of Material Components, use a `ThemeData` to specify default colors for various aspects of the theme. Set the theme property in `MaterialApp` to the `ThemeData` object. The  [`Colors`](https://docs.flutter.io/flutter/material/Colors-class.html)
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 19]
```diff
@@ -0,0 +1,2411 @@
...
+{% endprettify %}
+
+
+
+### How do I add style themes?
+
+In React Native, common themes are defined for components in stylesheets and
+then used in components.
+
+In Flutter, you can add uniform styling for almost everything by defining the
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 20]
```diff
@@ -0,0 +1,2411 @@
...
+      ),
+    );
+  }
+```
+
+## State Management
+State is information that can be read synchronously when a widget is built or information that might change during the lifetime of a widget. To manage app state in Flutter, use a
+ [StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html) paired with a State object.
+
+### What are stateful and stateless widgets and how do I use them?
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 21]
```diff
@@ -0,0 +1,2411 @@
...
+        <StatusBar hidden={true} />
+        <Text>{display} </Text>
+        <Button onPress={this.toggleBlinkState} title={this.state.buttonText} />
+      </View>
+    );
+  }
+}
+```
+
+#### The StatelessWidget
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 22]
```diff
@@ -0,0 +1,2411 @@
...
+      child: new Text(
+        text,
+        textDirection: TextDirection.ltr,
+      ),
+    );
+  }
+}
+```
+
+In the previous example, you use the constructor of the `MyStatelessWidget` class
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 23]
```diff
@@ -0,0 +1,2411 @@
...
+to pass the `text`, which is marked as `final`. This class extends
+ `StatelessWidget`—it contains immutable data.
+
+The `build` method of a stateless widget is typically only called in three
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 1 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 3 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 4 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 2 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 5 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 6 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md (hunk 7 above)
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done:  How do I use Form widgets?

Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done - globally

Comment (czakdesign) [reply flutter-for-react-native/index.md]: I don't see this sentence in the file.

Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

File: flutter-for-react-native/index.md [hunk 24]
```diff
@@ -0,0 +1,2411 @@
...
+#### The StatelessWidget
+
+A `StatelessWidget` in Flutter is a widget that doesn't require a state change—it has no
+internal state to manage.
+
+ Stateless widgets are useful when the part of the user interface you are
+ describing does not depend on anything other than the configuration information
+ in the object itself and the [`BuildContext`](https://docs.flutter.io/flutter/widgets/BuildContext-class.html) in which the widget is inflated.
+
+ [Icons](https://docs.flutter.io/flutter/material/Icons-class.html), [IconButton](https://docs.flutter.io/flutter/material/IconButton-class.html), and [Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are examples of stateless widgets, which subclass [StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
```
[the remark this answers is not in the document]
Comment (czakdesign) [reply flutter-for-react-native/index.md]: Done

**Pull request merged**